helpers/logging/middleware.go

```diff
@@ -36,8 +54,17 @@ func AddLoggingHandler(next http.Handler) http.Handler {
 		reqId, _ := uuid.Uuid()
 		reqId = fmt.Sprintf("[req-id %s]", reqId)
 		l.Println(reqId, r.Method, r.URL)
-		lr := &logResponseWriter{w, 0}
+		lr := &logResponseWriter{w, 0, nil}
+		br := &logRequestReader{r.Body, make([]byte, r.ContentLength)}
+		r.Body = br
 		next.ServeHTTP(lr, r)
+
+		reqstr := fmt.Sprintf("%s", br.Body)
```

Comment:
Is reqstr (or resstr below) used anywhere other than the Debug call?  If not, it might be slightly more efficient to not bother with the Sprintf.  Maybe do the Sprintf inside the Debug function after we have determined that we are going to spit out the log message?